### PR TITLE
all: (mostly) automated clean up

### PR DESCRIPTION
**Pull request by gcurtis**
- Unset any user-defined Go environment variables `devbox.json` to make builds more reproducible. Clean up some of the build scripts.
- Remove `exp/slices` and `exp/maps` now that they're in the Go 1.21 stdlib.
- Run `find . -name '*.go' -exec gofumpt -extra -w {} \+` to format Go files.
- Upgrade dependencies that don't have any backwards incompatible changes.
- Organize direct vs. indirect modules and run `go mod tidy`.

### CODE REVIEW

File: devbox.json
Comment (mikeland73) [inline devbox.json:1]: Add `gofumpt` here script here. Then we can add CICD that runs it and errors out if there are any changes.

**Review by mikeland73 — APPROVED** [review]

**Comment by mikeland73** [comment]
Tests fails just look like timeouts. Probably the dependency changes caused longer build times.

File: devbox.json
Comment (gcurtis) [reply devbox.json:1]: Added a `devbox run fmt` script that runs in the lint job right before golangci-lint. If `CI` is set, it prints any unformatted files and exits with an error. When `CI` isn't set, it silently formats the files.

**Comment by gcurtis** [comment]
Merging because the CI error was due to a GitHub API rate limit (presumably due to force-pushing debug commits).

**Pull request merged**